beacon-light-client/plonky2/circuits_executables/src/crud/proof_storage/redis_proof_storage.rs

```diff
@@ -5,32 +5,47 @@ use redis::{aio::Connection, AsyncCommands};
 
 pub struct RedisStorage {
     connection: Connection,
+    protocol: String,
 }
 
 impl RedisStorage {
-    pub async fn new(connection_string: String) -> Result<RedisStorage, redis::RedisError> {
+    pub async fn new(
+        connection_string: String,
+        protocol: String,
+    ) -> Result<RedisStorage, redis::RedisError> {
         let client = redis::Client::open(connection_string)?;
         let connection = client.get_async_connection().await?;
-        Ok(RedisStorage { connection })
+        Ok(RedisStorage {
+            connection,
+            protocol,
+        })
     }
 }
 
 #[async_trait(?Send)]
 impl ProofStorage for RedisStorage {
     async fn get_proof(&mut self, identifier: String) -> Result<Vec<u8>> {
-        Ok(self.connection.get(&identifier).await?)
+        let key = format!("{}:{}", self.protocol, identifier);
+        Ok(self.connection.get(&key).await?)
     }
 
     async fn set_proof(&mut self, identifier: String, proof: &[u8]) -> Result<()> {
-        Ok(self.connection.set(&identifier, proof).await?)
+        let key = format!("{}:{}", self.protocol, identifier);
+        Ok(self.connection.set(&key, proof).await?)
     }
 
     async fn del_proof(&mut self, identifier: String) -> Result<()> {
-        Ok(self.connection.del(&identifier).await?)
+        let key = format!("{}:{}", self.protocol, identifier);
+        Ok(self.connection.del(&key).await?)
     }
 
     async fn get_keys_count(&mut self, pattern: String) -> usize {
-        let result: Vec<String> = self.connection.keys(pattern).await.unwrap();
+        let key = format!("{}:{}", self.protocol, pattern);
+        let result: Vec<String> = self.connection.keys(key).await.unwrap();
         result.len()
     }
+
+    fn set_protocol(&mut self, protocol: String) {
+        self.protocol = protocol;
+    }
```

Comment:
This doesn't need to be part of the proof storage api

Reply:
Even for file storage, dump all the proofs in a single directory. We are going for a key/value store, imo it should be flat

Reply:
Just prefix the balance proofs with the protocol name